src/h5web/visualizations/heatmap/AspectRatioToggler.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { FiToggleLeft, FiToggleRight } from 'react-icons/fi';
+import styles from './HeatmapToolbar.module.css';
+import { useHeatmapStore } from './store';
+
+function AspectRatioToggler(): JSX.Element {
+  const [keepAspectRatio, toggleAspectRatio] = useHeatmapStore(state => [
+    state.keepAspectRatio,
+    state.toggleAspectRatio,
+  ]);
+
+  return (
+    <button
+      className={styles.aspectRatioToggler}
+      type="button"
+      role="switch"
+      aria-checked={keepAspectRatio}
+      onClick={() => {
+        toggleAspectRatio();
+      }}
+    >
+      {keepAspectRatio ? <FiToggleRight /> : <FiToggleLeft />}
+      Keep aspect ratio
+    </button>
+  );
+}
+
+export default AspectRatioToggler;
```

Comment:
Just thinking that it may be worth implementing a generic `Toggler` component that takes a `className`, a `label`, a `value` and an `onChange` prop. This would localise selections from the store in the toolbar component (single hook call with shallow comparison)